temperature_regulator/server_thread.py

```diff
@@ -0,0 +1,55 @@
+from threading import Thread, Lock
+import socket
+import select
+from typing import Tuple
+
+class ServerThread(Thread):
+
+    RECV_MSG_MAX_SIZE = 1024
+    TEXT_ENCODING = 'UTF-8'
+    CLIENT_TIMEOUT = 3
+
+    _print_lock = Lock()
+
+    def __init__(self, id: int, connection_socket: socket.socket, client_address_pair: Tuple[str, str]):
+        Thread.__init__(self)
+        if connection_socket:
```

Comment:
można to zapisać w konwecji "Fail first" wtedy jest jedno wcięcie mniej:
```python
        if not connection_socket:
           raise RuntimeError("No socket specified")
        
        self._connection_socket = connection_socket
        self._connection_socket.settimeout(ServerThread.CLIENT_TIMEOUT)
        self._client_address, self._client_port = client_address_pair
        self._client_port = int(self._client_port)  # Change str to int
        self._id = id
```
ale to moja mała opinia